readme.adoc

```diff
@@ -1,37 +1,162 @@
+** Configure the program by providing your credentials.
++
+[source, bash]
+----
+aws configure
+----
+
+** Check the `aws` credentials are present in the right files in your home directory.
++
+[source, bash]
+----
+vim $HOME/.aws/credentials
+----
+
+. Get the latest version of the **CDP CLI**.
+** https://docs.cloudera.com/cdp/latest/cli/topics/mc-installing-cdp-client.html[Install CDP CLI].
+
+. If this is the first time you are installing the CDP CLI, you will need to configure the program by providing the right credentials.
++
+====
+Getting the right credentials for CDP (public or private) might be an offline step.
+Once you have the right credentials continue below.
+See https://docs.cloudera.com/cdp/latest/cli/topics/mc-cli-generating-an-api-access-key.html[CDP API Keys] for further details.
+====
+
+** Configure the program by providing your credentials.
++
+[source, bash]
+----
+cdp configure
+----
+
+** Check the `cdp` credentials are present in the right files in your home directory.
++
+[source, bash]
+----
+vim $HOME/.cdp/credentials
+----
+
+. Check you have a generated SSH keypair for your local profile.
+** Read the https://www.ssh.com/academy/ssh/keygen[SSH Keygen How-To]
+
+. Check you have a properly configured SSH Agent.
+** Read the https://www.ssh.com/academy/ssh/keygen#adding-the-key-to-ssh-agent[SSH Agent How-To]
+
+TIP: Put AWS CLI and CDP CLI programs in your `$PATH` to make these two programs easily accessible.
+
+=== Setup
+
+. Clone this repository
++
+[source, bash]
+----
+git clone https://github.com/cloudera-labs/cloudera-deploy.git
+----
+WARNING: Be careful to not modify any of the files in the project as a user of the software. The vast majority of changes are managed through configurations provided to these project files.
+
+. Check that **Docker** is running.
+** https://docs.docker.com/docker-for-windows/[For Windows]
+** https://docs.docker.com/docker-for-mac/[For Macs]
+** Linux users, use `systemd` (or another init system)
+
+. Run the `quickstart.sh` entrypoint script.
++
+[source, bash]
+----
+cd cloudera-deploy
+chmod +x quickstart.sh
+./quickstart.sh
+----
+
+. Confirm that you have the orange _cldr_ prompt.
++
+====
+This is your interactive Ansible Runner environment and provides builtin access to the relevant dependencies for CDP.
+====
++
+IMPORTANT: Do _NOT_ run the example definition until you have made the changes below.
+
+. Modify your local `cloudera-deploy` user profile. This is present in your `$HOME` directory under `~/.config/cloudera-deploy/profiles/default`.
++
+[source, bash]
+----
+vim ~/.config/cloudera-deploy/profiles/default
+----
+Properties to change: 
+** Recommended
+*** *admin_password:* Note the minimal password requirements.
+*** *name_prefix:* Follow the notes mentioned.
+*** *infra_type:* The valid values are `aws`, `gcp`, `azure`.
+*** *infra_region:* Region is dependent on the value provided in `infra_type`. This is _not_ checked later during the run.
```

Comment:
Rather than 'this is not checked later during the run' it would be better to say 'please ensure you provide a valid region for the infra_type'

Reply:
Changed and added a tooltip